base/src/external_transition.rs

```diff
@@ -27,8 +28,42 @@ pub struct ExternalTransition {
     pub current_protocol_version: ProtocolVersion,
     /// Proposed protocol version
     pub proposed_protocol_version_opt: Option<ProtocolVersion>,
-    /// Callback used for validating external transition received over the network.
-    pub validation_callback: (),
+}
+
+use crate::network_types::v1::ExternalTransitionV1;
+
+impl From<ExternalTransitionV1> for ExternalTransition {
+    fn from(t: ExternalTransitionV1) -> Self {
+        let t = t.0.inner();
+        Self {
+            protocol_state: t.protocol_state.into(),
+            protocol_state_proof: t.protocol_state_proof,
+            staged_ledger_diff: t.staged_ledger_diff.into(),
+            delta_transition_chain_proof: t.delta_transition_chain_proof,
+            current_protocol_version: t.current_protocol_version,
+            proposed_protocol_version_opt: t.proposed_protocol_version_opt,
+        }
+    }
+}
+
+impl Into<ExternalTransitionV1> for ExternalTransition {
```

Comment:
IIRC it's recommended to implement From trait instead, that way u get Into impl for free